### DISCUSSION TITLE
How do this effort differ from truck?

### DISCUSSION THREAD

**Discussion by mattiasgronlund** [hannobraun/fornjot#202] [General]
Have you looked into https://github.com/ricosjp/truck?

I get a feeling that much of the groundwork needed like CAD kernel is the same as for truck, making me think that some cooperation could be beneficial for both projects.
/Mattias

**Comment by hannobraun** [comment]
Hey @mattiasgronlund, good question!

I'm aware of Truck, and I've considered switching to it. I decided to not do that (yet).

A CAD kernel is a *really* big and important dependency for a CAD application. To a large extent, it defines what the application can be. That means choosing one is a big decision.

I'm far from an expert on CAD kernels, I'm just learning as I go. I can't fully judge whether Truck is a super-solid option that would fit all my needs, or whether I'd run into limitations sooner or later. The best way I learn is hands-on, so I decided to continue my own effort and revisit this decision at a later date. Then, I'll have a much better idea of what Fornjot needs, and will be in a better position to evaluate Truck.

Another way to reach the same point would be to dive into Truck instead, by starting to use it immediately. I can't tell which way would be better, so I chose the one that seems more fun and satisfying.

**Comment by thawkins** [comment]
Have you considered defining an interface to your kernel that is compatible with truck, so that fornjot can be built with either?

**Reply by hannobraun** [reply]
In fact, I haven't!

This might be valuable, but I suspect that it either would be a big effort (with ongoing maintenance cost). Might be interesting to explore later, but I don't think I'll attempt this any time soon.
